R/run_trial.R

```diff
@@ -152,65 +152,36 @@ run_trial <- function(
 
     # post-processing to get evaluation metrics
     if (!is.null(design$evaluation)){
-      # AUC metrics
-      if (any(names(design$evaluation) %in% target_types_auc)){
-        auc_evals <- names(design$evaluation)[names(design$evaluation) %in% target_types_auc]
-        for (design_type in auc_evals){
-          tmp_eval_design <- list(
-            type = design_type,
-            scheme = design$evaluation[[design_type]]
-          )
-          auc_eval <- calc_auc_from_regimen(
-            regimen = res$final_regimen,
-            parameters = pars_true_i, # true patient parameters
-            model = design$sim$model,
-            target_design = tmp_eval_design,
-            covariates = covs
-          )
-          eval_time <- get_sampling_times_from_scheme(
-            tmp_eval_design$scheme,
-            res$final_regimen
-          )
-          eval_exposure <- rbind(
-            eval_exposure,
-            data.frame(
-              id = i,
-              time = eval_time,
-              value = auc_eval,
-              type = tmp_eval_design$type
-            )
-          )
+      for (design_type in names(design$evaluation)){
+        tmp_eval_design <- list(
+          type = design_type,
+          scheme = design$evaluation[[design_type]]
+        )
+        if (design_type %in% target_types_auc){
+          f_calc <- calc_auc_from_regimen
+        } else if (design_type %in% target_types_conc){
+          f_calc <- calc_concentration_from_regimen
         }
-      }
-      # concentration metrics
-      if (any(names(design$evaluation) %in% target_types_conc)){
-        conc_evals <- names(design$evaluation)[names(design$evaluation) %in% target_types_conc]
-        for (design_type in conc_evals){
-          tmp_eval_design <- list(
-            type = design_type,
-            scheme = design$evaluation[[design_type]]
+        exposure_metric <- f_calc(
```

Comment:
what happens if `design_type` is neither in `target_types_auc` nor `target_types_conc`? let's make sure that isn't a blocking error, or that we return an informative message if that's the case!